### ISSUE TITLE
Feature request for missions: waypoints that require you to land on a planet, not just enter the system

### ISSUE THREAD

**Issue by Pointedstick** [endless-sky/endless-sky#407]
This would add a lot of flexibility for missions that are currently difficult to implement. This feature should also provide standard events and triggers that can be executed when you land on each of the planet waypoints.

This would be useful for a number of different circumstances, from the common case of retrieving cargo or passengers on another system and returning to the planet where the missions as offered, to more complicated situations where you need to land on multiple planets such as the part of the Free Worlds mission where you're looking for the planet used as a nuclear testing site.

**Comment by endless-sky** [comment]
This can be implemented just by making separate missions for those planets.
Each of those missions would only be offered when the main mission is
active, and they would set a condition flag that is necessary for the main
mission to complete.

**Comment by Pointedstick** [comment]
True, but that's a lot more complicated than having planet-based waypoints.

**Comment by omagasohe** [comment]
EV had a travel to X pick up cargo and take it to Y as a mission choice. I'd like it for job missions, there should be a large amounts of jobs types. the pick up and deliver are missing.

**Comment by Pointedstick** [comment]
That's exactly my thinking, @omagasohe. I am adding a bunch of new job types to spice things up a bit and really missing this feature.

**Comment by endless-sky** [comment]
I've finally added support for this, via a new "stopover" keyword (d46d099). It works like this:

```
mission "Stopover Test"
	name "Pick up <commodity>"
	description "Pick up <cargo> on <stopovers>, then return to <destination>, where you will be paid <payment>."
	job
	repeat
	source
		government Republic "Free Worlds" Syndicate Neutral
	stopover
		distance 4 6
		government Republic "Free Worlds" Syndicate Neutral
	cargo random 25 2 .05
	on stopover
		dialog
			"The dock workers load up your cargo of <commodity>, to be delivered to <destination>."
	on complete
		payment 10000 1000
		dialog
			"The dock workers unload your shipment of <commodity> and pay you <payment>."
```

The only problem is that the payment calculation is based on the number of jumps to the destination, and does not take into account how far you will have to travel for any stopovers. In the example above I addressed that by having the mission pay 1000 credits per cargo ton (instead of 150) and limiting the stopover to being in a relatively small distance range.

**Comment by Pointedstick** [comment]
Yay!!!!

**Comment by Pointedstick** [comment]
Does this mean that deadlines won't work because it will calculate the time between the source and the destination without taking into account the travel time to and from the stopover planet?

**Comment by endless-sky** [comment]
Included in release 0.8.10.
